### PR TITLE
Remove use of IApplicationEnvironment

### PR DESCRIPTION
**Pull request by pranavkm**
Hosting is no longer going to add these services to the container and we'll be removing the interfaces entirely (only keeping the static helpers)

### CODE REVIEW

**Comment by pranavkm** [comment]
cc @divega - who'd be the right person to review this?
cc @davidfowl

**Comment by divega** [comment]
This affects commands so assigning to @natemcmaster.

**Comment by natemcmaster** [comment]
> we'll be removing the interfaces entirely

@pranavkm Can you point to a PR or issue in Hosting so we know when this change has been taken in to Hosting?

**Comment by pranavkm** [comment]
Hosting work item: https://github.com/aspnet/Hosting/pull/723

**Comment by natemcmaster** [comment]
@pranavkm  this is going into RC2?

**Comment by pranavkm** [comment]
Yup.

File: src/Microsoft.EntityFrameworkCore.Commands/Design/Internal/ApplicationEnvironment.cs [hunk 1]
```diff
@@ -1,28 +0,0 @@
...
-
-namespace Microsoft.EntityFrameworkCore.Design.Internal
-{
-    public class ApplicationEnvironment : IApplicationEnvironment
-    {
-        public ApplicationEnvironment([NotNull] IApplicationEnvironment applicationEnvironment)
-        {
-            Check.NotNull(applicationEnvironment, nameof(applicationEnvironment));
-
-            ApplicationBasePath = applicationEnvironment.ApplicationBasePath;
```
Comment (natemcmaster) [inline src/Microsoft.EntityFrameworkCore.Commands/Design/Internal/ApplicationEnvironment.cs]: What is going to replace this?

Comment (pranavkm) [inline src/Microsoft.EntityFrameworkCore.Commands/Design/Internal/ApplicationEnvironment.cs]: The only reference to this was in `StartupInvoker` which was removed. There's already code that sets up `HostingEnvironment` with the right `ContentRoot`

**Comment by natemcmaster** [comment]
This is going to be problematic, unless the static helpers are settable. The path of `DefaultApplicationEnvironment.ApplicationBasePath` at design time is the nuget cache. Anyone that ues Configurate.SetBasePath to ApplicationBasePath is not going to be able to resolve config files at design-time.

**Comment by natemcmaster** [comment]
:shipit: 

Note to self: when this is in, we need to manually test and verify this doesn't break `--startup-project` on commands.

**Comment by davidfowl** [comment]
Looks good :shipit:

**Pull request closed without merging**